internal/api/rbac/roles.go

```diff
@@ -713,41 +705,47 @@ func RoleToResources(
 	return sa, role, rb, nil
 }
 
-func replaceClaimAnnotation(sa *corev1.ServiceAccount, key string, values []string) {
-	slices.Sort(values)
-	values = slices.Compact(values)
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
+func replaceClaimAnnotation(sa *corev1.ServiceAccount, prefix string, values []rbacapi.Claim) {
+	for _, claim := range values {
+		slices.Sort(claim.Values)
+		claim.Values = slices.Compact(claim.Values)
+		if sa.Annotations == nil {
+			sa.Annotations = map[string]string{}
+		}
+		sa.Annotations[prefix+claim.Name] = strings.Join(claim.Values, ",")
 	}
-	sa.Annotations[key] = strings.Join(values, ",")
 }
 
-func amendClaimAnnotation(sa *corev1.ServiceAccount, key string, values []string) {
-	existing := sa.Annotations[key]
-	if existing != "" {
-		values = append(strings.Split(existing, ","), values...)
-	}
-	slices.Sort(values)
-	values = slices.Compact(values)
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
+func amendClaimAnnotation(sa *corev1.ServiceAccount, prefix string, values []rbacapi.Claim) {
+	for _, claim := range values {
+		existing := sa.Annotations[prefix+claim.Name]
+		if existing != "" {
+			claim.Values = append(strings.Split(existing, ","), claim.Values...)
+		}
```

Comment:
Super-nit: Slightly more idiomatic:

```suggestion
		if existing, exists := sa.Annotations[prefix+claim.Name]; exists {
			claim.Values = append(strings.Split(existing, ","), claim.Values...)
		}
```

Might apply in some related functions as well.

Reply:
That's much better, fixed.